docs/consume-packages/Package-References-in-Project-Files.md

```diff
@@ -170,6 +170,20 @@ Conditions can also be applied at the `ItemGroup` level and will apply to all ch
 </ItemGroup>
 ```
 
+When you have many targets, it may be better to match ranges of TFMs, in which case you can use [`IsTargetFrameworkCompatible`](/visualstudio/msbuild/property-functions#msbuild-targetframework-and-targetplatform-functions):
+```xml
+<ItemGroup>
+    <!-- reference 8.0 System.Text.Json when targeting things older than .NET 8 -->
+    <PackageReference Include="System.Text.Json"  Version="8.0.5" Condition=" !$([MSBuild]::IsTargetFrameworkCompatible('$(TargetFramework)', 'net8.0')) " />
```

Comment:
This is probably me overthinking it because of https://learn.microsoft.com/en-us/nuget/consume-packages/package-references-in-project-files#prunepackagereference, but the idea is that we generally want to make sure people are not referencing platform packages unless they're absolutely required. 
For example, there's many System.Text.Json references when customers target .NET and that's unnecessary. 

So I'm thinking that we can maybe use a Contoso named package, or  if we're sticking with System.Text.Json we can use: GetTargetFrameworkIdentifier. 

Maybe we end up with 2 examples, 1 using `IsTargetFrameworkCompatible`, while the other one uses `GetTargetFrameworkIdentifier`

Reply:
As in `$([MSBuild]::GetTargetFrameworkIdentifier('$(TargetFramework)')) != '.NETCoreApp'`?

Reply:
Yes.

Reply:
Since the .NET SDK sets property `TargetFrameworkIdentifier`, and MSBuild evaluates properties top to bottom before items, doing a TFI condition is easier by using `Condition=" '$(TargetFrameworkIdentifier)' != '.NETCoreApp' "`, rather than calling the instrinsic function and passing the target framework as an argument.

Reply:
Done, but I can't leave the top of this section unchanged since it's now self-contradictory: "[only the `TargetFramework` variable is supported](https://github.com/YoshiRulz/docs.microsoft.com-nuget/blob/5e37700e67e6c199c3daa814a7cd942895a2a50e/docs/consume-packages/Package-References-in-Project-Files.md?plain=1#L146)"